Orange/widgets/evaluate/owrocanalysis.py

```diff
@@ -601,6 +604,57 @@ def _setup_plot(self):
                 warning = "All ROC curves are undefined"
         self.warning(warning)
 
+    def _on_mouse_moved(self, pos):
+        target = self.target_index
+        selected = self.selected_classifiers
+        curves = [(clf_idx, self.plot_curves(target, clf_idx))
+                  for clf_idx in selected]  # type: List[Tuple[int, plot_curves]]
+
+        for clf_idx, crv in curves:
+            if self.roc_averaging == OWROCAnalysis.Merge:
+                curve = crv.merge()
+            elif self.roc_averaging == OWROCAnalysis.Vertical:
+                curve = crv.avg_vertical()
+            elif self.roc_averaging == OWROCAnalysis.Threshold:
+                curve = crv.avg_threshold()
+            else:
+                # currently not implemented for 'Show Individual Curves'
+                return
+
+            sp = curve.curve_item.childItems()[0]  # type: pg.ScatterPlotItem
+            act_pos = sp.mapFromScene(pos)
+            pts = sp.pointsAt(act_pos)
+
+            if len(pts) > 0:
+                mouse_pt = pts[0].pos()
+                if self._tooltip_cache:
+                    cache_pt, cache_thresh, cache_clf, cache_ave = self._tooltip_cache
+                    if numpy.linalg.norm(mouse_pt - cache_pt) < 10e-6 and clf_idx == cache_clf \
+                            and cache_ave == self.roc_averaging:
+                        if not QToolTip.isVisible():
+                            QToolTip.showText(QCursor.pos(),
+                                              "Threshold: {:.3f}".format(cache_thresh))
+
+                        return
+                    else:
+                        QToolTip.showText(QCursor.pos(), "")
+                        self._tooltip_cache = None
+
+                curve_pts = curve.curve.points
+
+                # Find closest point on curve and display it
+                idx_closest = numpy.argmin(
+                    [numpy.linalg.norm(mouse_pt - [curve_pts.fpr[idx], curve_pts.tpr[idx]])
+                     for idx in range(len(curve_pts.thresholds))])
```

Comment:
This loop could be vectorized
```python
roc_points = numpy.column_stack((curve_pts.fpr, curve_pts.tpr))
query_pt = numpy.array([[mouse_pt.x(), mouse_pt.y()]])
diff = roc_points - query_pt
idx_closest = numpy.argmin(numpy.linalg.norm(diff, axis=1))
```
(... although a slightly moot point as the `sp.pointsAt` call above already runs through all the points in a python loop)

Reply:
True - fixed.  
I've also made another minor change: now QCursor.pos() is directly used everywhere instead of using it directly in some cases and in the other cases saving it and then using the variable (just a consistency fix).